Login as Guest

Nishitbaria opened the pull request:
## Related Issue
<!--
Provide information about the issue or bug that is being addressed.
-->

Closes: #138 

## Description of Changes
This pull request introduces the addition of a Guest login button to both the home page (index.php) and the login page (Login.php) of the project. The button functions as a pop-up or modal triggered by a click event. The pop-up contains two input fields: "Original Links" and "Custom Name." Additionally, it includes two buttons: "Random Number" and "Save Link."

Upon clicking the "Save Link" button, the application performs Link validations and then also check availability of Custom Name  . If all validations pass successfully, the system generates a shortened link. A success alert is then displayed to inform the user about the successful creation of the shortened link. Furthermore, a new modal is shown, which includes the shortened link and a "Copy" button.

To maintain a record of the shortened links, the application also saves the data in a database. Each entry is associated with a unique number ("short") and linked to the username "shorty."



## Checklist: 

<!--
Mark the checkboxes to indicate completion. Example: 
- [x] My code follows the style guidelines of this project.
-->

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [ ] I have included comments in areas that may be difficult to understand.
- [ ] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.


## Screenshots

|           |           |
|:---------:|:---------:|
| **2.1**   | **2.2**   |
| ![Issue2 1](https://github.com/apoorvaron/Shorty/assets/85815172/1e431e33-ab88-4d95-a503-63d265390893) | ![Issue2 2](https://github.com/apoorvaron/Shorty/assets/85815172/6eed69ed-fea0-4c89-a87d-06ededbd9622) |
|           |           |
| **2.3**   | **2.4**   |
| ![Issue2 3](https://github.com/apoorvaron/Shorty/assets/85815172/3c4fcb3f-d675-4a40-8b29-8ea9b2d255c4) | ![Issue2 4](https://github.com/apoorvaron/Shorty/assets/85815172/7bb89dd9-b6f1-49bc-a301-5bb98653d084) |
|           |           |


Uploading New video.mp4…


Please provide any necessary screenshots to illustrate the changes made.

Nishitbaria commented:
hey @apoorvaron  Please review my PR and Merge it

apoorvaron reviewed (changes requested):
There are some changes 

1) Expected better modal UI (Transparent type modal + background blur type) i want you to change styling 

<img width="379" alt="Screenshot 2023-06-22 at 2 07 11 AM" src="https://github.com/apoorvaron/Shorty/assets/83576954/ad0eefe0-70f8-4b5b-be6f-a7e5c0f3fdd9">


2) Maintain colour of buttons( like initially light blue & on hover it changes) 

<img width="551" alt="Screenshot 2023-06-25 at 5 58 24 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/555600c6-2af0-4951-b200-070385d5ec99">

3) Add Guest to Home Page too

4) Just append the shortened link with copy button in this pop up itself
<img width="546" alt="Screenshot 2023-06-25 at 6 00 35 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/b8eb8319-b5bb-4741-8f32-dbf36fb21069">

5) While i am testing , the favicon disappears

<img width="175" alt="Screenshot 2023-06-25 at 6 01 41 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/477bc56c-f080-4c80-89b2-a47e8c400e1b">

Nishitbaria commented:
Ok I m on it @apoorvaron 

Is the placement of Create Custom Button Appropriate?

Let me Know any Changes needed 

![Capture](https://github.com/apoorvaron/Shorty/assets/85815172/39df40c5-11fc-4e69-acf0-bb40b10ea884)

apoorvaron commented:
@Nishitbaria But Now , if i short the link using this 👇, then UI will be ? overlapped?
<img width="649" alt="Screenshot 2023-06-25 at 6 48 56 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/5e71c384-6def-4f83-a397-016c5991f098">

apoorvaron commented:
Or you can just add it to the top right and make Other Dropdown or something

Nishitbaria commented:
@apoorvaron This is current Scenario there is not overlapping so should change them placement of button or not 

and I have Suggestion that we should keep only a single button which provides both functionality of with random name and other with custom name

![image](https://github.com/apoorvaron/Shorty/assets/85815172/414ec575-0cfc-4f5a-a776-0c989bc6c757)

apoorvaron commented:
@Nishitbaria About suggestion , then what will be the full Flow ?

Nishitbaria commented:
> @Nishitbaria About suggestion , then what will be the full Flow ?

1.When user clicks on shorten button 
2.Modal opens,where he can short link by custom name or by random number
3.shorten link will be there in modal,so user can copy it

apoorvaron commented:
@Nishitbaria ig current situation is fine for home page

As you did in login page like pop up stuff, do same in home page too. Just add a Guest button in top right 

click on Guest (Home Page / Login  :: Same Pop up :: No duplicate code :: Make common files for pop up ) -> Same pop up appear -> then do requested changes

Nishitbaria commented:
![image](https://github.com/apoorvaron/Shorty/assets/85815172/813bff72-f7c6-4518-90fc-fe6379d5a7ff)
![WhatsApp Image 2023-06-26 at 03 17 16](https://github.com/apoorvaron/Shorty/assets/85815172/3a2a11a8-7b45-46a0-a37e-8794009d300d)

@apoorvaron This is type of blur i implemented ...

apoorvaron commented:
@Nishitbaria its not looking good , i suggest you to try like 
click on guest -> pop up appear + background changes to https://app.xobin.com/static/onboardingv2/images/new_pat_crop.svg

And try to create exactly same pop up , i have added the website link 
( https://app.xobin.com/signin?_ga=2.26136829.1390229748.1687752808-1863597800.1687752808 )

apoorvaron commented:
@Nishitbaria or you redirect user to another page, but don't compilcate it for user


like Click on guest -> redirect to another page Guest.php* -> do stuff and this page contain all required links or make same as front page but it contains only needed things like header+footer+same blue theme+ centralised input boxes

Nishitbaria commented:
@apoorvaron Ok I got it

Now Please Clear me some things ok...
1. Theme should equivalent to homepage or newlink.php ?
2. I have to implement Session and previous shorten links pages ?

Awaiting for your reply

apoorvaron commented:
@Nishitbaria 
1) Equivalent to home page 
2) dont implement session or login 
just click on guest -> redirect to another page , which contains all the fields that you implemented in pop up with shorted link and copy btn

Nishitbaria commented:
@apoorvaron Got It

Nishitbaria commented:
https://github.com/apoorvaron/Shorty/assets/85815172/38199931-cf27-4134-a055-422fb6bb121c

This are changes i made

Nishitbaria commented:
Please go through this @apoorvaron

apoorvaron commented:
@Nishitbaria 
1) Guest.php does not contain Guest Button (use same header.php)
2) if successfully created then no sweet alert is needed, if error occur then show sweet alert
3) prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field 
4) Remove "Custom Link" or reduce the font-size 
5) Use common files for header and footer and chatbotsidebar

Nishitbaria commented:
> @Nishitbaria
> 
> 1. Guest.php does not contain Guest Button (use same header.php)
> 2. if successfully created then no sweet alert is needed, if error occur then show sweet alert
> 3. prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field
> 4. Remove "Custom Link" or reduce the font-size
> 5. Use common files for header and footer and chatbotsidebar
> 6. Share mobile view too
> 7. don't clear fields if successfully created link

**@apoorvaron** Thanks for your review!

1. The Guest button is not visible in Guest.php due to a condition added in header.php. This condition was implemented to hide the Guest Button as its active page.
```php
<?php
if (!isset($isGuest)) {
    echo '
    <li>
        <a class="get-a-quote" href="./guest">Guest Login</a>
    </li>';
}
?>
```
2. Task completed.
3. Like this as Refered in  image ? ![image](https://github.com/apoorvaron/Shorty/assets/85815172/b83c6781-0e91-4075-8026-7adcb09c6775)
4. Font size has been reduced.
5. The header and footer files are common as they are included from includes folder.
6. Mobile view screenshots:

| Mobile View 1                                               | Mobile View 2                                           |
| ----------------------------------------------------------- | ------------------------------------------------------- |
| ![Image1](https://github.com/apoorvaron/Shorty/assets/85815172/408f2da2-2ed7-4bf4-8913-298e7b217906) | ![Image2](https://github.com/apoorvaron/Shorty/assets/85815172/85e0283b-9cdf-44e0-abda-8f81fc7bc782) |
| ![Image3](https://github.com/apoorvaron/Shorty/assets/85815172/c5ea3a1d-2b80-4359-9a41-170e4a76a707) | 

7. Currently working on this feature and will update soon.

apoorvaron commented:
@Nishitbaria 
1) don't hide it. Also, Button Tex "Guest Login" to "Guest"
3) Yes,
6) Remove this description from guest page (desktop view as well as mobile view) not heading "Shorty"

<img width="637" alt="Screenshot 2023-06-28 at 10 38 01 AM" src="https://github.com/apoorvaron/Shorty/assets/83576954/0962cfcb-27ff-4562-b2ac-43851c6d9dac">

VinayakVispute commented:
@apoorvaron 
Ok and previous i was using Post request for Guest Custom link but now its almost converted to ajax request
because post request was rising some issues like Refresh of page and form clearing and sometime post errors arises when we press back button to come to Guest page from other pages

Nishitbaria commented:
@apoorvaron
Ok and previous i was using Post request for Guest Custom link but now its almost converted to ajax request
because post request was rising some issues like Refresh of page and form clearing and sometime post errors arises when we press back button to come to Guest page from other pages

apoorvaron commented:
> 

ok, you can do whatever you want, but code should work 😂

Nishitbaria commented:
> @apoorvaron Ok and previous i was using Post request for Guest Custom link but now its almost converted to ajax request because post request was rising some issues like Refresh of page and form clearing and sometime post errors arises when we press back button to come to Guest page from other pages

I have successfully implemented Ajax Requst and now the problem of Form Clearance and Refresh of page and post request issues solved.

Below mentioned Changes
1. Added this function in \includes\footer.php
```php
  function generateCustomShorty() {
            let originalLink = document.querySelector('#originalLink').value;
            let shortenLink = document.querySelector('#shortenLink').value;
            console.log(originalLink);
            console.log(shortenLink);
            let avail = shortenLink;
            // Check if the original link has been previously shortened
            let urllink = "./admin/dBconn/api.php?q=alreadyShortenCustom&shortenLink=" + shortenLink;
            $.ajax({
                url: urllink,
                method: 'GET',
                dataType: 'JSON',
                success: function (data) {
                    console.log("this is datt", data);
                    // If the link is previously shortened
                    if (data) {
                        swal("Custom Name Not available", "", "error");
                        return;

                    } else {
                        // If the link is not previously shortened
                    }

                    let regexp = /^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff]-*)*[a-z\u00a1-\uffff]+)(?:\.(?:[a-z\u00a1-\uffff]-*)*[a-z\u00a1-\uffff]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/;
                    let ifValidLink = regexp.test(originalLink);
                    if (ifValidLink) {
                        console.log("Testing", ifValidLink);
                        let generateShorty = document.querySelector('#generateShorty');
                        let full_shortlink = "<?php echo $siteName ?>";
                        generateShorty.innerHTML = `
                    <form class="form-search d-flex align-items-stretch mb-3" data-aos="fade-up" data-aos-delay="200">
                        <input type="text" id="shortInput" disabled style="font-size: 0.9rem;" disabled class="form-control" placeholder="" value="<?php echo $siteName ?>${avail}"/>
                        <input class="btn btn-primary" type="button" onclick="copy()" id="copyBtn" value="Copy">
                    </form>
                `;

                        let originalLink = document.querySelector('#originalLink').value;

                        var formData = new FormData();
                        formData.append('originalLink', originalLink);

                        let url = "./admin/dBconn/api.php/?q=shorty&shortenLink=" + avail;
                        $.ajax({
                            type: "POST",
                            url: "./admin/dBconn/api.php/?q=shorty&shortenLink=" + avail,
                            data: formData,
                            cache: false,
                            processData: false,
                            contentType: false,
                            success: function (data) {
                                console.log("success");
                            },
                            error: function (xhr, status, error) {
                                console.log("No");
                            },
                        });
                    } else {
                        swal("Enter Valid URL !!", "", "error");
                    }
                },
                error: function (xhr, status, error) {
                    console.log("No");
                },
            });
        }
```
2. Added this in api.php
```javascript
function alreadyShortenCustom()
{
    $database = new Database();
    $db = $database->connect();

    $shortenLink = $_GET['shortenLink'];
    $sql = "SELECT * FROM links WHERE shortenLink='" . $shortenLink . "'";
    $result = $db->query($sql);

    if ($result && $result->num_rows > 0) {
        echo json_encode(true);
    } else {
        echo json_encode(false);
    }
}
;
```

3.  Added this condition
```javascript
case 'alreadyShortenCustom';
        alreadyShortenCustom();
        break;
```

Nishitbaria commented:
> > 
> 
> ok, you can do whatever you want, but code should work 😂

Yes its working 😂😂

apoorvaron commented:
@Nishitbaria should i review now , or are you working ?

apoorvaron commented:
> > > 
> > 
> > 
> > ok, you can do whatever you want, but code should work 😂
> 
> Yes its working 😂😂
> 
> It first checks if custom name is available. If not, display a SweetAlert. Should I hide the shorten link pop-up if the user tries to shorten the link again second time after once and the custom name is unavailable?
> 
> This one ![image](https://user-images.githubusercontent.com/85815172/249366562-1561217f-acde-4f06-988f-e5c8edc1b178.png)

No , because multiple users can short the same link. Example yt link

apoorvaron commented:
> @Nishitbaria
> 
> 1. don't hide it. Also, Button Tex "Guest Login" to "Guest"
> 2. Yes,
> 3. Remove this description from guest page (desktop view as well as mobile view) not heading "Shorty" or you can center the things like this
> 
> <img alt="Screenshot 2023-06-28 at 11 27 37 AM" width="809" src="https://user-images.githubusercontent.com/83576954/249368792-53c4bed2-eabf-46f9-9bb4-adfcb600ee24.png"> <img alt="Screenshot 2023-06-28 at 10 38 01 AM" width="637" src="https://user-images.githubusercontent.com/83576954/249358010-0962cfcb-27ff-4562-b2ac-43851c6d9dac.png">
> 7. Rest : don't clear fields if successfully created link

@Nishitbaria updated

Nishitbaria commented:
Wait i have of add one condition that if the user doesnt enter Custom Name and do this one

> prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field

apoorvaron commented:
> Wait i have of add one condition that if the user doesnt enter Custom Name and do this one
> 
> > prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field

@Nishitbaria don't get you 
User have to enter custom name in guest page, Custom name is mandatory to be filled to short the link

Nishitbaria commented:
> > Wait i have of add one condition that if the user doesnt enter Custom Name and do this one
> > > prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field
> 
> @Nishitbaria don't get you User have to enter custom name in guest page, Custom name is mandatory to be filled to short the link

Yes since i was testing i removed that condition now its done
1. It Checks if empty or single space then Error
2. if the custom name contains like this `nishit Insta` then i will converted to `nishit_Insta`

apoorvaron commented:
okay

Nishitbaria commented:
> Wait i have of add one condition that if the user doesnt enter Custom Name and do this one
> 
> > prepend "http://shrty.rf.gd/" from siteName.php file before custom Name field

Both are Done 

https://github.com/apoorvaron/Shorty/assets/85815172/16c1cae7-0484-4f32-a091-d83210a5ef9c

## Mobile View for New Created Form (With Prepend) 
![WhatsApp Image 2023-06-28 at 13 33 28](https://github.com/apoorvaron/Shorty/assets/85815172/13119eb0-09b4-4124-abb6-5d18ea964c01)

Nishitbaria commented:
@apoorvaron You can Review it 
This prepend thing took my lots of time 😂😂

apoorvaron commented:
@Nishitbaria 
just reduce these spaces so that no scrolling should not happen and bold the domain ($siteName) and then i will merge it 

<img width="1440" alt="Screenshot 2023-06-28 at 1 54 26 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/53222109-ea5b-4a5b-8217-c13fcf1db3eb">

Nishitbaria commented:
@apoorvaron Done can you check it now

apoorvaron commented:
@Nishitbaria Great work man 🎉

Nishitbaria commented:
@apoorvaron Thanks Man Its means a lot

apoorvaron commented:
@Nishitbaria 

1)Why these files are changed ? you did changes in these files too ? n why ?

<img width="109" alt="Screenshot 2023-06-28 at 2 44 50 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/9556152e-414b-4feb-b5b0-e44947df36c3">

2) For my phone (one plus 9RT) mobile view is not perfect as below 

<img width="380" alt="Screenshot 2023-06-28 at 2 46 50 PM" src="https://github.com/apoorvaron/Shorty/assets/83576954/ea1f0981-1de8-4c36-9e12-06eaa7b9577b">

Nishitbaria commented:
For 1st part I have gone through my files and its fine no changes are there are i have removed all my code added it may be from previous commit which i have removed my code or while copy i may have added or removed blank lines

dont worry i have gone through and everything is fine

but this one alert i added have removed

Nishitbaria commented:
ok for part 2 i am fixing it

Nishitbaria commented:
First request: Dont add multiple labels just remove level2 label  and keep level3 label
second request: Can I  create a issue  for improving responsiveness for mobile view

apoorvaron commented:
@Nishitbaria Yes , create another issue
